src/Module/OStatus/Salmon.php

```diff
@@ -66,12 +66,12 @@ protected function rawContent(array $request = [])
 	{
 		$xml = Network::postdata();
 
-		$this->logger->debug('New Salmon', ['nickname' => $this->parameters['nickname'], 'xml' => $xml]);
-
 		// Despite having a route with a mandatory nickname parameter, this method can also be called from
 		// \Friendica\Module\DFRN\Notify->post where the same parameter is optional 🤷‍
 		$nickname = $this->parameters['nickname'] ?? '';
 
+		$this->logger->debug('New Salmon', ['nickname' => $nickname, 'xml' => $xml]);
```

Comment:
I guess that the corresponding warning occurs because of the call to the DFRN endpoint that calls this endpoint here. I guess that the fix has to go deeper.

Reply:
Yes, there already is a comment about it, I just suppressed the warning message.

Reply:
Doesn't just the user from the DFRN endpoint be passed to this?

Reply:
No. The problem is that in the DFRN endpoint, the username is optional, and when it's missing, no user is fetched, but the Salmon endpoint is called. (see `src/Module/DFRN/Notify.php:73`)

Reply:
Okay. I'm not 100% sure what this endpoint was meant to do, but I have got the idea that it was possibly meant for subscribing/unsubscribing to users using some OStatus magic with DFRN. I suggest to simply drop calling it from the DFRN endpoint.

Reply:
Gladly.

Reply:
Done.